Main svc

SinitsaBogdan opened the pull request.

SinitsaBogdan committed:
refactor :  Mapper and NameClass exeptions

SinitsaBogdan committed:
create : modules in main

SinitsaBogdan committed:
create : main-dto

SinitsaBogdan committed:
create : main-server - user

SinitsaBogdan committed:
create : main-server - util

SinitsaBogdan committed:
create : main-server - category

SinitsaBogdan committed:
refactor : module statistics - EndpointServiceImpl.java

SinitsaBogdan committed:
create : main-server - event

SinitsaBogdan committed:
create : main-server - location

SinitsaBogdan committed:
create : main-server - compilation

SinitsaBogdan committed:
create : main-server - participationRequest

SinitsaBogdan committed:
refactor : modules main-dto and Mappers

SinitsaBogdan committed:
refactor : module naiming

SinitsaBogdan committed:
refactor : project

SinitsaBogdan committed:
refactor : controllers

SinitsaBogdan committed:
refactor : controllers

SinitsaBogdan committed:
refactor : services

SinitsaBogdan committed:
refactor : services

SinitsaBogdan committed:
refactor : models

SinitsaBogdan committed:
refactor : models

SinitsaBogdan committed:
refactor : dto

SinitsaBogdan committed:
refactor : dto

SinitsaBogdan committed:
refactor : project

SinitsaBogdan committed:
refactor : project

SinitsaBogdan committed:
refactor : repo

SinitsaBogdan committed:
refactor : project

SinitsaBogdan committed:
Merge branch 'main_svc' of https://github.com/SinitsaBogdan/java-explore-with-me into main_svc

SinitsaBogdan committed:
refactor : client

SinitsaBogdan committed:
refactory : project

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/category/controller/CategoryPublicController.java:
```diff
@@ -0,0 +1,37 @@
+
+    @GetMapping
+    public List<CategoryDto> getAll(
+            @Valid @RequestParam(defaultValue = "0") @Min(0) int from,
```
Напротив примитивов не требуется ставить @valid, достаточно Validated над контроллером

SinitsaBogdan replied:
DONE

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/category/service/CategoryServiceImpl.java:
```diff
@@ -0,0 +1,69 @@
+
+    @Override
+    @Transactional
+    public CategoryDto patch(long catId, CategoryDto categoryDto) {
```
На мой взгляд, было бы логичнее назвать метод update, так как patch скорее подходит для api

SinitsaBogdan replied:
DONE

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/controller/EventPublicController.java:
```diff
@@ -0,0 +1,91 @@
+
+    @GetMapping
+    public List<EventShortDto> getAll(
+            @RequestParam(defaultValue = "") String text,
```
Было бы хорошо также валидировать размер строки, так как он ограничен по размерам

SinitsaBogdan replied:
DONE

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/controller/EventPublicController.java:
```diff
@@ -0,0 +1,91 @@
+            @PathVariable long eventId,
+            HttpServletRequest request
+    ) {
+        System.out.println("EventPublicController - client.saveHit");
```
Для вывода информации лучше использовать логи

SinitsaBogdan replied:
да, видимо пропустил когда удалял.

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/category/service/CategoryServiceImpl.java:
```diff
@@ -0,0 +1,69 @@
+    @Override
+    @Transactional
+    public void delete(long catId) {
+        Category category = categoryRepository.findById(catId)
```
category нигде не используется, поэтому не стоит её заводить

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/category/service/CategoryServiceImpl.java:
```diff
@@ -0,0 +1,69 @@
+    @Override
+    @Transactional
+    public void delete(long catId) {
+        Category category = categoryRepository.findById(catId)
```
Также перед удалением необходимо проверить, есть ли в этой категории евенты, и если есть, то выбрасывать ошибку с информацией об этом

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/compilation/service/CompilationServiceImpl.java:
```diff
@@ -0,0 +1,83 @@
+        Compilation compilation = compilationRepository.findById(compId)
+                .orElseThrow(() -> new NotFoundException("Сборник с id " + compId + " не найден!"));
+
+        if (compilationRequestUpdateDto.getEvents() != null) compilation.setEvents(
```
Было бы хорошо проверять не только на null, но и на пустоту списка, так как в этом случае также нет смысла обращаться к бд

SinitsaBogdan replied:
Done

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/compilation/service/CompilationServiceImpl.java:
```diff
@@ -0,0 +1,83 @@
+    @Override
+    @Transactional
+    public void delete(long compId) {
+        Compilation compilation = compilationRepository.findById(compId)
```
Также нет особого смысла заводить compilation, так как она нигде не используется

SinitsaBogdan replied:
Done

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,399 @@
+            eventList = eventList.stream()
+                    .filter(
+                            event -> event.getParticipantLimit().equals(0)
+                            || event.getParticipantLimit() < participationRequestRepository.countByEventIdAndStatus(
```
Не стоит обращаться к бд в цикле, так как это значительно влияет на производительность всего приложения. Также не стоит допускать ситуации, когда в цикле идет обращение к сервису статистики

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,399 @@
+                            .findFirst();
+                    dto.setViews(matchingStats.map(ViewEndpointDto::getHits).orElse(0L));
+                })
+                .peek(dto -> dto.setConfirmedRequests(participationRequestRepository.countByEventIdAndStatus(dto.getId(), ParticipationRequestState.CONFIRMED)))
```
Также не стоит обращаться к бд в цикле

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,399 @@
+    public List<EventShortDto> getAllByInitiator(long userId, int from, int size) {
+        Pageable pageable = PageRequest.of(from, size);
+        Page<Event> page = eventRepository.findAllByInitiatorId(userId, pageable);
+        if (page.isEmpty()) return new ArrayList<>();
```
Лучше возвращать Collections.emptyList

SinitsaBogdan replied:
Done

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-stats-service/stats-client/src/main/java/ru/practicum/client/StatisticsClient.java:
```diff
@@ -30,8 +33,8 @@ public void saveHit(EndpointHitDto endpointHitDto) {
-                .retrieve()
-                .bodyToMono(Void.class);
+                .body(BodyInserters.fromValue(endpointHitDto))
+                .exchange()
```
Методы, которые объявлены как устаревшие лучше не использовать

SinitsaBogdan replied:
откатил к предыдущему варианту
были какие то проблемы с сохранением, в этом спринте что то переделывал в клиенте

SinitsaBogdan replied:
+ ко всему этому у меня по какой то причине не подсвечиваются эти методы как устаревшие
никаких предупреждений IDEA не выдает

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/user/controller/UserPrivateEventsController.java:
```diff
@@ -0,0 +1,82 @@
+    @GetMapping
+    public List<EventShortDto> getAllEventShort(
+            @PathVariable long userId,
+            @Valid @RequestParam(defaultValue = "0") @Min(0) int from,
```
Не обязательно ставить valid напротив переменных

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/user/mapper/UserMapper.java:
```diff
@@ -0,0 +1,36 @@
+public final class UserMapper {
+
+    private UserMapper() {
+        throw new CustomException("Это служебный класс, и его экземпляр не может быть создан");
```
Хорошая практика

SinitsaBogdan replied:
👍

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/dto/EventNewDto.java:
```diff
@@ -0,0 +1,51 @@
+    private Long category;
+
+    @NotNull
+    private LocationDto location;
```
Можно валидировать при помощи @Min и @Max для того, чтобы задать диапазон допустимых значений

SinitsaBogdan replied:
Done

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,399 @@
+                            .findFirst();
+                    dto.setViews(matchingStats.map(ViewEndpointDto::getHits).orElse(0L));
+                })
+                .peek(dto -> dto.setConfirmedRequests(participationRequestRepository.countByEventIdAndStatus(dto.getId(), ParticipationRequestState.CONFIRMED)))
```
Аналогично, лучше избегать ситуаций когда к бд идет обращение в цикле

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/dto/EventResultUpdateStatusDto.java:
```diff
@@ -0,0 +1,20 @@
+@AllArgsConstructor
+public class EventResultUpdateStatusDto {
+
+    List<ParticipationRequestDto> confirmedRequests;
```
У переменных DTO классов должны быть модификаторы доступа

SinitsaBogdan replied:
Done

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/dto/EventShortDto.java:
```diff
@@ -0,0 +1,31 @@
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = Constants.DATE_TEMPLATE)
+    private LocalDateTime eventDate;
+
+    private Boolean paid;
```
(предложение)paid может иметь только два состояния, а в Boolean можно передать null, поэтому было бы хорошо сделать его примитивом

SinitsaBogdan replied:
действительно, вроде и мелочь, но все же логично, даже не задумался об этом
по стандарту просто использую классы типов

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/event/dto/EventRequestUpdateStatusDto.java:
```diff
@@ -0,0 +1,25 @@
+public class EventRequestUpdateStatusDto {
+
+    @NotNull
+    private List<Long> requestIds;
```
Можно использовать set для исключения дублей

SinitsaBogdan replied:
тоже верно, о Set начинаешь думать только по факту возникновения такой необходимости в результате

Done

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/compilation/dto/CompilationDto.java:
```diff
@@ -0,0 +1,21 @@
+
+    private Long id;
+    private String title;
+    private Boolean pinned;
```
Аналогично, лучше использовать примитив

SinitsaBogdan replied:
Done

SinitsaBogdan replied:
Done

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/compilation/dto/CompilationRequestUpdateDto.java:
```diff
@@ -0,0 +1,22 @@
+    @Size(max = 50)
+    private String title;
+
+    private Boolean pinned;
```
Также можно использовать примитив

SinitsaBogdan replied:
Done

SinitsaBogdan replied:
Done

SinitsaBogdan committed:
refactor : controllers (riviev)

rev : Напротив примитивов не требуется ставить @Valid, достаточно Validated над контроллером.

SinitsaBogdan committed:
refactor : service (riviev)

rev : На мой взгляд, было бы логичнее назвать метод update, так как patch скорее подходит для api

SinitsaBogdan committed:
refactor : project (riviev)

rev : Было бы хорошо также валидировать размер строки, так как он ограничен по размерам

SinitsaBogdan committed:
refactor : project (riviev)

rev : Для вывода информации лучше использовать логи

SinitsaBogdan committed:
refactor : CategoryServiceImpl

rev : category нигде не используется, поэтому не стоит её заводить

SinitsaBogdan committed:
refactor : CategoryServiceImpl and EventRepository

rev : Также перед удалением необходимо проверить, есть ли в этой категории евенты, и если есть, то выбрасывать ошибку с информацией об этом

SinitsaBogdan committed:
Update UserPrivateEventsController.java

SinitsaBogdan committed:
refactor : CompilationServiceImpl.java

rev : Было бы хорошо проверять не только на null, но и на пустоту списка, так как в этом случае также нет смысла обращаться к бд

SinitsaBogdan committed:
refactor : CompilationServiceImpl.java

rev : Также нет особого смысла заводить compilation, так как она нигде не используется

SinitsaBogdan committed:
refactor : EventServiceImpl.java

rev : Лучше возвращать Collections.emptyList

SinitsaBogdan committed:
refactor : EventNewDto.java

rev : Можно валидировать при помощи @min и @max для того, чтобы задать диапазон допустимых значений

SinitsaBogdan committed:
refactor : EventResultUpdateStatusDto.java

rev : У переменных DTO классов должны быть модификаторы доступа

SinitsaBogdan committed:
refactor : EventShortDto.java

rev : (предложение) paid может иметь только два состояния, а в Boolean можно передать null, поэтому было бы хорошо сделать его примитивом

SinitsaBogdan committed:
refactor : EventRequestUpdateStatusDto.java

rev : Можно использовать set для исключения дублей

SinitsaBogdan committed:
refactor : EventServiceImpl.java.java

rev : Можно использовать set для исключения дублей

SinitsaBogdan committed:
refactor : CompilationDto.java

rev : Аналогично, лучше использовать примитив

SinitsaBogdan committed:
refactor : CompilationRequestUpdateDto.java

rev : Также можно использовать примитив

SinitsaBogdan committed:
refactor : StatisticsClient.java

rev : Также можно использовать примитив

SinitsaBogdan committed:
refactor : изменение логики запроса событий и формирование результата

rev : Не стоит обращаться к бд в цикле, так как это значительно влияет на производительность всего приложения. Также не стоит допускать ситуации, когда в цикле идет обращение к сервису статистики
